middleware/auth.ts

```diff
@@ -0,0 +1,56 @@
+import { NextFunction, Request, Response } from 'express';
+import * as jwt from "jsonwebtoken";
+import { ErrorMessage } from '../models/errors';
+import { UserEntity } from '../models/user.entity';
+
+export async function verifyToken(req: Request, res: Response, next: NextFunction) {
+    const authHeader: string | undefined = req.headers.authorization;
+
+    if (!authHeader) {
+        return res
+            .status(401)
+            .send({
+                "data": null,
+                "error": {
+                    "message": ErrorMessage.Unauthorized
+                }
+            });
+    }
+
+    try {
+        const [tokenType, token] = authHeader.split(' ');
+
+        if (tokenType !== 'Bearer') {
+            return res.status(403).send({
+                "data": null,
+                "error": {
+                    "message": ErrorMessage.Forbidden
+                }
+            });
+        }
+
+        const user = jwt.verify(token, process.env.TOKEN_KEY!) as UserEntity;
```

Comment:
Avoid using the non-null assertion operator `!`. Instead, validate `TOKEN_KEY` during the startup of your application and throw an error if it's not set

Reply:
Changed. Thanks)